sonar-text-plugin/src/main/java/org/sonar/plugins/common/TextAndSecretsSensor.java

```diff
@@ -76,42 +67,32 @@ public void describe(SensorDescriptor descriptor) {
 
   @Override
   public void execute(SensorContext sensorContext) {
+    // Retrieve list of checks
     List<Check> activeChecks = getActiveChecks();
     durationStatistics = new DurationStatistics(sensorContext.config());
     initializeSpecificationBasedChecks(activeChecks);
     if (activeChecks.isEmpty()) {
       return;
     }
 
-    NotBinaryFilePredicate notBinaryFilePredicate = binaryFilePredicate(sensorContext);
-    FilePredicate filePredicate = isSonarLintContext(sensorContext) || analyzeAllFiles(sensorContext)
-      ? notBinaryFilePredicate
-      : LANGUAGE_FILE_PREDICATE;
+    // Retrieve list of files to analyse using the right FilePredicate
+    boolean analyzeAllFiles = isSonarLintContext(sensorContext) || analyzeAllFiles(sensorContext);
+    var notBinaryFilePredicate = binaryFilePredicate(sensorContext);
+    var textFilePredicate = textFilePredicate(sensorContext);
+    FilePredicate filePredicate = analyzeAllFiles ? notBinaryFilePredicate : CombinedFilePredicate.any(LANGUAGE_FILE_PREDICATE, textFilePredicate);
```

Comment:
CombinedFilePredicate can be replaced by `OR` on the predicate itself. Docker has an example for it